eox_nelp/pearson_vue/api/v1/views.py

```diff
@@ -201,6 +202,30 @@ def create(self, request, *args, **kwargs):
 
         return response
 
+    def get_object(self):
+        """
+        Retrieves the latest object for the given candidate and course.
+
+        This method filters the queryset to find objects matching the current user
+        and the course specified by the lookup field in the URL. If no objects are found,
+        it raises a 404 error. If objects are found, it returns the most recently created object.
+
+        Returns:
+            object: The most recently created object that matches the given candidate and course.
+
+        Raises:
+            Http404: If no objects match the given candidate and course.
+        """
+        objects = self.get_queryset().filter(
+            candidate=self.request.user,
```

Comment:
> I was testing this, and I would check the reason of the candidate using only the `request.user`. I mean this API is by Bearer so the the user would be the same as the third-party- application. So this would only allow to view courses of the application user. Eg. my application or credentials of the Api are attached to **cms** user. So after I create a resultNotification related cms_user I could get data of ![image](https://private-user-images.githubusercontent.com/51926076/349246745-59fe28c1-0975-4d76-8e1b-bbefe6d44f56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNjEwMDQsIm5iZiI6MTcyMTE2MDcwNCwicGF0aCI6Ii81MTkyNjA3Ni8zNDkyNDY3NDUtNTlmZTI4YzEtMDk3NS00ZDc2LThlMWItYmJlZmU2ZDQ0ZjU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDIwMTE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkODRmNjU0YmRjNzg0ODMxOWFmNjg2YmQyOTg5M2JkNzE4NWVhZjRmYjI4ZTczZjgxNzBiOGM0MDliNGQyYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lGUYBKo4PuQW3Asi8844hpUz1kMDuhMeRbB_b9Fjy3o) ![image](https://private-user-images.githubusercontent.com/51926076/349247049-ce755c21-a8dd-4f13-b766-cc62028edd6c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNjEwMDQsIm5iZiI6MTcyMTE2MDcwNCwicGF0aCI6Ii81MTkyNjA3Ni8zNDkyNDcwNDktY2U3NTVjMjEtYThkZC00ZjEzLWI3NjYtY2M2MjAyOGVkZDZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDIwMTE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhMTFiNTQ5NDUyZjYxNzcxMmQ3MGQyZjQ0NmUxMDBkZjc1ZmQ1NTUzMTMxZjVhMzI5ZmJlZjZkNzllZDE3YjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JFPTGuKMYrQCHIeNDcTUZSMzVJRt8PyHtVdvzby3C_0)

you are right, that is the purpose of these changes

Reply:
But why is this useful? only to check the pearsonbearer application user?

Reply:
the bearer application user, the jwt application user  and finally any user with a jwt token(which is not associated to a Django toolkit application )